python/pyspark/ml/param/__init__.py

```diff
@@ -375,6 +375,18 @@ def copy(self, extra=None):
         that._defaultParamMap = {}
         return self._copyValues(that, extra)
 
+    def set(self, param, value):
+        """
```

Comment:
I included this function because the getAndSetParams function in DefaultParamsReader required a way to set a parameter for an instance given the param object and the value of the param. There wasn't any helper functionality in the Params class that did this so I added the set() function from Scala. Is it okay to leave it for this purpose or is there a better way of handling this case?